assets/pages/regular-contributions/helpers/ajax.js

```diff
@@ -32,41 +32,74 @@ type ContributionRequest = {
   billingPeriod: BillingPeriod,
 };
 
+type PaymentFieldName = 'baid' | 'stripeToken' | 'directDebitData';
+
+type DirectDebitDetails= {|
+  accountHolderName: string,
+  sortCode: string,
+  accountNumber: string
+|};
+
 type RegularContribFields = {|
   firstName: ?string,
   lastName: ?string,
   country: IsoCountry,
   state?: UsState,
   contribution: ContributionRequest,
   paymentFields: {
-    stripeToken: string,
-  },
+    [PaymentFieldName]: string,
+  } | DirectDebitDetails,
   ophanIds: OphanIds,
   referrerAcquisitionData: ReferrerAcquisitionData,
   supportAbTests: AcquisitionABTest[],
 |};
 
-type PaymentField = 'baid' | 'stripeToken' | 'directDebitData';
-
-
 // ----- Functions ----- //
 
 const isUserValid = (user: UserState) =>
   user.firstName !== null && user.firstName !== undefined &&
   user.lastName !== null && user.lastName !== undefined &&
   user.email !== null && user.email !== undefined;
 
+const getPaymentFields =
+  (
+    token?: string,
+    accountNumber?: string,
+    sortCode?: string,
+    accountHolderName?: string,
+    paymentFieldName: string,
+  ) => {
```

Comment:
Is it possible to give this a return type?

Reply:
Added it here: https://github.com/guardian/support-frontend/pull/460/commits/c1e072a7a5e2167d1a2ee25a6a9aedca59ef5846